app/controllers/moments_controller.rb

```diff
@@ -127,6 +127,6 @@ def associated_strategies
     Strategy.where(user: @viewers).each do |strategy|
       strategy_ids << strategy.id if strategy.viewer?(current_user)
     end
-    Strategy.where(id: strategy_ids).order(created_at: :desc)
+    Strategy.is_visible.where(id: strategy_ids).order(created_at: :desc)
```

Comment:
An edge case that we should account for is what happens when we create a Moment with a Strategy/Category/Mood OR a Strategy with a Category that is visible when created, but is later edited to be not visible?

Example:

1. Create a Moment with a Strategy that is visible

![image](https://user-images.githubusercontent.com/3010728/195976394-8b6827fe-7a16-41f0-b6db-a7b69c25481c.png)

2. Edit the Strategy and make it not visible

![image](https://user-images.githubusercontent.com/3010728/195976425-44986376-f528-4776-87a2-30887735286d.png)

3. Edit the Moment and the Strategy is not in the list

![image](https://user-images.githubusercontent.com/3010728/195976493-66f8ecfb-caad-413a-932f-75bbb166ecf7.png)

---

I think the expected behaviour for 3) should be that we can view the selected Strategy. If we uncheck the box for it and save our changes, then we can no longer add it back.

Reply:
I am unable to understand the problem statement, please explain again.

Reply:
I think if it's already tagged, but the tag is made not visible, it should still be visible to the post (or parent like you said) that it's being used in.

Reply:
I think I understand now. It's late but I should have a commit tomorrow, I will have to do away with the scope Concern though.